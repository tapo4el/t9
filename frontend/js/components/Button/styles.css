.t9-app .button {
    background: #e0e1e2 none;
    cursor: pointer;
    height: 50px;
    flex:1 1;
    font: 14px Lato,'Helvetica Neue',Arial,Helvetica,sans-serif;
    border: 1px solid #ffffff;
    outline: 0;
}
.t9-app .button:hover {
    background: #cacbcd;
}
.t9-app .button:first-child {
    border-top-left-radius: 10px;
    border-bottom-left-radius: 10px;
}
.t9-app .button:last-child {
    border-top-right-radius: 10px;
    border-bottom-right-radius: 10px;
}
