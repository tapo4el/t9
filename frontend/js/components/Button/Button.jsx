import React from 'react';
import { func, string } from 'prop-types';

import './styles.css';

class Button extends React.PureComponent {
    static propTypes = {
        text: string,
        value: string.isRequired,
        onClickHandler: func.isRequired,
    };

    render () {
        const { text, value, onClickHandler } = this.props;
        return (
            <button className="button" onClick={() => onClickHandler(value)}>
                {value}
                <div>{text}</div>
            </button>
        );
    }
}

export default Button;
