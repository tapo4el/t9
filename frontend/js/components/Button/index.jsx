import React from 'react';
import './styles.css';

class Button extends React.PureComponent {
    render () {
        const { text, value, addNumber } = this.props;
        return (
            <button className="button" onClick={() => addNumber(value)}>
                {value}
                <div>{text}</div>
            </button>
        );
    }
}

export default Button;
