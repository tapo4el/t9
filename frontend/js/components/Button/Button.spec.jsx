import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Button from './Button';

describe('Button', () => {
    it('should render correctly', () => {
        const component = renderer.create(
            <Button value="2" onClickHandler={() => {}} text="abc" />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should work correctly', () => {
        const spy = jest.fn();

        const wrapper = mount(<Button value="2" onClickHandler={spy} text="abc" />);
        wrapper.find('.button').simulate('click');
        expect(spy).toHaveBeenCalled();
        expect(spy.mock.calls[0]).toMatchSnapshot();
    });
});
