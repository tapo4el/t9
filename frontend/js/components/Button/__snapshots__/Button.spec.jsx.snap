// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
>
  2
  <div>
    abc
  </div>
</button>
`;

exports[`Button should work correctly 1`] = `
Array [
  "2",
]
`;
